cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -248,14 +249,15 @@ private TetheringLaunchMessage createLaunchPayload(RuntimeJobInfo runtimeJobInfo
     Collection<? extends LocalFile> localFiles = runtimeJobInfo.getLocalizeFiles();
     for (String fileName : builder.getFileNames()) {
       LocalFile localFile = localFiles.stream().filter(file -> file.getName().equals(fileName))
-          .findFirst().orElseThrow(() -> new IllegalStateException("Cannot find file" + fileName));
+          .findFirst().orElseThrow(() -> new IllegalStateException("Cannot find file " + fileName));
       builder.addLocalizeFiles(fileName, getLocalFileAsCompressedBytes(localFile));
     }
 
     for (String prefix : SELECT_CCONF_FIELDS) {
       String prefixRegex = "^" + prefix.replace(".", "\\.");
       builder.addCConfEntries(cConf.getValByRegex(prefixRegex));
     }
+    builder.addCConfEntry(Constants.AppFabric.PROGRAM_JVM_OPTS, runtimeJobInfo.getJvmOpts());
```

Comment:
I'm still not clear why this is needed when we're already sending the program options over. Doesn't it still go through the DistributedProgramRunner eventually on the tethered side where the options get added to the TwillPreparer?

In other words, what breaks if we just do the change in the DistributedProgramRunner? Does the tethered side go through a different code path for some reason?

Reply:
Oh you're right, thanks for pointing that out! I was testing with this change (add jvm opts to twill preparer from program options) applied only on the CDF side, so the updated jvmp opts didn't take effect.
When this fix is applied on both sides, the cConf update is unnecessary - fixed now. Thank you!